libraries/GCS_MAVLink/MAVLink_routing.h

```diff
@@ -32,8 +32,11 @@ class MAVLink_routing
     /*
       send a MAVLink message to all components with this vehicle's system id
       This is a no-op if no routes to components have been learned
+
+      msgid here is the mavlink message ID, pkt is a pointer to a
+      mavlink message structure (e.g. a mavlink_command_long_t)
     */
-    void send_to_components(const mavlink_message_t &msg);
+    void send_to_components(uint32_t msgid, const char *pkt);
```

Comment:
I'd like a length arg, and chec

Reply:
I've added a an assert in SITL and we pass through the smaller of the passed packet length and the maximum message length.